arkane/explorer.py

```diff
@@ -294,17 +295,29 @@ def execute(self, outputFile, plot, format='pdf', print_summary=True, speciesLis
 
                     for P in Plist:
                         if self.flux_tol != 0.0:
-                            rxns = network.get_rate_filtered_reactions(T, P, self.flux_tol)
-                            if rxnSet is not None:
-                                rxnSet &= set(rxns)
+                            products = network.get_rate_filtered_products(T, P, self.flux_tol)
+                            products = [tuple(x) for x in products]
+                            if productSet is not None:
+                                productSet &= set(products)
                             else:
-                                rxnSet = set(rxns)
+                                productSet = set(products)
 
-                logging.info('removing reactions during reduction:')
-                for rxn in rxnSet:
-                    logging.info(rxn)
 
-                network.remove_reactions(reaction_model, list(rxnSet))
+                if rxnSet:
+                    logging.info('removing reactions during reduction:')
+                    for rxn in rxnSet:
+                        logging.info(rxn)
+                if productSet:
+                    logging.info('removing products during reduction:')
+                    for prod in productSet:
+                        logging.info([x.label for x in prod])
+
+                if rxnSet:
```

Comment:
maybe combine these conditions into the ones above?

Reply:
done